### PR TITLE
updated the architecture doc

### PR DESCRIPTION
**Pull request by yaronp68**

### CODE REVIEW

**Review by dwdougherty — CHANGES_REQUESTED** [review]
Lots of small changes.

File: content/rdi/architecture.md [hunk 1]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
 aliases: 
 ---
 
-Redis Data Integration is a product for data ingestion from existing systems of record (mainly databases) into Redis Enterprise.
-You can think of Redis Data Integration as a streaming ELT process:
+Redis Data Integration (RDI) is a product that helps Redis Enterprise users ingest and export data in near real time
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:12]: done

File: content/rdi/architecture.md [hunk 2]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
 
-- Data is **E**xtracted from the source database using [Debezium](https://debezium.io/) connectors.
+- End to end solution - No need for additional tools and integrations
+- Capture Data Change (CDC) included
+- Covers most popular databases as sources & targets
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:16]: done

File: content/rdi/architecture.md [hunk 3]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+- Declarative data mapping and transformations - no custom code needed
+- Data delivery guaranteed (at least once)
+- Zero downtime day 2 operations (Reconfigure & Upgrade)
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:19]: done

File: content/rdi/architecture.md [hunk 4]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+
+Redis Data Integration currently supports 2 use cases:
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:21]: done

File: content/rdi/architecture.md [hunk 5]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+
+- Ingest (Cache prefetch).
+- Write Behind.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:24]: done

File: content/rdi/architecture.md [hunk 6]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+## Ingest functionality and architecture
+
+You can think of RDI Ingest as a streaming ELT process:
+
+- Data is **E**xtracted from the source database using [Debezium Server](https://debezium.io/) connectors.
 - Data is then **L**oaded into Redis Data Integration, a Redis database instance that keeps the data in [Redis streams](https://redis.io/docs/manual/data-types/streams/) alongside required metadata.
 - Data is then **T**ransformed using provided [RedisGears recipes](https://developer.redis.com/howtos/redisgears/) and written to the target Redis DB.
 
-Redis Data Integration using Debezium works in 2 modes:
+Redis Data Integration using Debezium Server works in 2 modes:
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:34]: done

File: content/rdi/architecture.md [hunk 7]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+
+#### Mode mapping
+
+RDI supports conversion of a database record into the following Redis types:
+
+- Hash
+- Set
+- Sorted Set
+- Stream
+- Json
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:51]: done

File: content/rdi/architecture.md [hunk 8]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+- String
+
+Each column in the record is automatically formatted as a number or string. See the [data types list]({{<relref "/rdi/reference/data-types-conversion">}}) for the exact mappings.
+
+#### Additional transformations
 
-Currently Redis Data Integration supports transforming data from RDBMS row into a [Redis Hash](https://redis.io/docs/manual/data-types/#hashes).
-In the short term we will add support for RDBMS row into [RedisJSON](https://redis.io/docs/stack/json/) document.
-See the [data types list]({{<relref "/rdi/reference/data-types-conversion">}}) for the exact mappings.
+RDI supports declarative transformations to further manipulate the data, including denormalization of data from few records into a single Json key in Redis.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:58]: done

File: content/rdi/architecture.md [hunk 9]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
 
-## Architecture and components
+Red the data transformation section of these docs to learn more.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:60]: done

File: content/rdi/architecture.md [hunk 10]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+### Architecture and components
 
 ![Detailed Redis Data Integration architecture](/images/rdi/redis-di.png)
 
-## Feeders
+#### Feeders
 
-Currently the only source for Redis Data Integration is the [Debezium Redis Sink Connector](https://debezium.io/documentation/reference/stable/operations/debezium-server.html#_redis_stream).
-Other sources and SDK will be added in later versions.
+Debezium Server has a Redis sink that writes data & state to RDI streams.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:68]: done

File: content/rdi/architecture.md [hunk 11]
```diff
@@ -9,59 +9,89 @@ headerRange: "[2]"
...
+In addition RDI works with [Arcion](arcion.io) commercial CDC that provides additional set of sources.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:69]: done

File: content/rdi/architecture.md [hunk 12]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
 - Running on 2 VMs in an active/passive setup. The failover orchestration can be done by tools such as [Pacemaker](https://clusterlabs.org/pacemaker/doc/).
 - Running as a Kubernetes pod where Kubernetes watches the pod liveliness and readiness and acts to recover nonfunctional pod
   It is important to note that Redis Data Integration does not keep any state on Debezium. All state is kept in Redis Data Integration.
+
+## Write behind functionality and architecture
+
+RDI write behind allows integration between Redis Enterprise (as the source of changes to data) to downstream databases or datastores.
+RDI captures the changes to a selected set of key-patterns in Redis keyspace and asynchronsly writes them in small batches to the downstream database, so the application doesn't need to take care of remodeling the data or managing the connection with the downstream database.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:134]: done

File: content/rdi/architecture.md [hunk 13]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+
+RDI write behind can normalize a key in Redis to several records in one or more tables at the target.
+To learn more about write behind declarative jobs and renormalization read the [write behind quick start guide]({{<relref="/rdi/quickstart/write-behind-guide.md">}})
+
+### Write behind topology
+
+RDI CLI and engine comes in one edition that can run both ingest and write behind. However, the topology for write behind is different.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:141]: done

File: content/rdi/architecture.md [hunk 14]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+RDI engine is installed on Redis database containing the application data and not on a separate staging Redis database. RDI streams and control plain adds a small footprint of few hundreds of MBs to the Redis database. In the write behind topology, RDI process in parallel on each shard and establishes a single connection from each shard to the downstream database.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:142]: done

File: content/rdi/architecture.md [hunk 15]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+
+### model translation
+
+RDI write behind can track changes to the following Redis types:
+
+- Hash
+- Json
+- Set
+- Sorted Set
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:151]: done

File: content/rdi/architecture.md [hunk 16]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
+
+Unlike the ingest scenario, write behind has no default behavior for model translation. It is up to the user to create a declarative job, specifying the mapping between Redis key and target database record.
```
[the remark this answers is not in the document]
Comment (yaronp68) [reply content/rdi/architecture.md:153]: done

**Review by dwdougherty — APPROVED** [review]

File: content/rdi/architecture.md [hunk 17]
```diff
@@ -97,3 +127,28 @@ Debezium Server should run in warm High Availability topology in one of the foll
...
 - Running as a Kubernetes pod where Kubernetes watches the pod liveliness and readiness and acts to recover nonfunctional pod
   It is important to note that Redis Data Integration does not keep any state on Debezium. All state is kept in Redis Data Integration.
+
+## Write behind functionality and architecture
+
+RDI write behind allows integration between Redis Enterprise (as the source of changes to data) to downstream databases or datastores.
+RDI captures the changes to a selected set of key-patterns in a Redis keyspace and asynchronously writes them in small batches to the downstream database, so the application doesn't need to take care of remodeling the data or managing the connection with the downstream database.
+
+RDI write behind can normalize a key in Redis to several records in one or more tables at the target.
+To learn more about write behind declarative jobs and renormalization read the [write behind quick start guide]({{<relref="/rdi/quickstart/write-behind-guide.md">}})
```
Comment (dwdougherty) [inline content/rdi/architecture.md:137]: One more fix... the relref shortcode syntax was incorrect (no = sign).

```suggestion
To learn more about write behind declarative jobs and renormalization read the [write behind quick start guide]({{< relref "/rdi/quickstart/write-behind-guide.md" >}})
```

Comment (yaronp68) [reply content/rdi/architecture.md:137]: done

**Pull request merged**